### DISCUSSION TITLE
Option to make retry and exit on pause screen hold-to-confirm

### DISCUSSION THREAD

**Discussion by SonoSooS** [ppy/osu#18709] [Ideas]
I have to pause a lot, and it has happened to me a lot of times that in high 90% progress on a beatmap I pause, but proceed to accidentally instantly click retry or exit in various methods, like the window resizing, the mouse cursor teleporting after a window resize, mouse flying out of my hand, or accidentally having a twitch on the mouse button as soon as I pause.

It would be also great if the timeout for this was separately controllable, or not controllable at all from the hold-to-confirm delay time in the settings, as when I must pause, I must pause instantly, so I have hold-to-confirm down at 0ms.

It would be even better if it was possible to remove those buttons all together, and have dedicated keyboard keybinding instead to hold to exit, or hold the already existing quick retry button to retry.

**Comment by peppy** [comment]
They should probably be switched to dangerous buttons.

**Reply by respektive** [reply]
A potential issue I can see with the retry button being dangerous is that it can be very annoying to retry at the beginning of the map if you get a miss for example and just wanna quick restart. Perhaps it should only be dangerous if you're say over 10% into the map or so.

**Reply by SonoSooS** [reply]
I would say "it should be a few seconds instead due to on longer maps 10% meaning a lot of minutes", but then I realized that this problem could be solved just by editing the quick restart time, and using the quick restart shortcut instead.

Although a few seconds after the first hitobject would be better, I think.
